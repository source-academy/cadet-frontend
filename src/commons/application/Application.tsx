import { decompressFromEncodedURIComponent } from 'lz-string';
import * as qs from 'query-string';
import * as React from 'react';
import { Redirect, Route, RouteComponentProps, Switch } from 'react-router';

import { Variant } from 'js-slang/dist/types';

import Academy from '../../pages/academy/AcademyContainer';
import Contributors from '../../pages/contributors/Contributors';
import Login from '../../pages/login/LoginContainer';
import MissionControlContainer from '../../pages/missionControl/MissionControlContainer';
import NotFound from '../../pages/notFound/NotFound';
import Playground from '../../pages/playground/PlaygroundContainer';
import SourcecastContainer from '../../pages/sourcecast/SourcecastContainer';
import NavigationBar from '../navigationBar/NavigationBar';
import { stringParamToInt } from '../utils/ParamParseHelper';
import { Role, sourceLanguages } from './ApplicationTypes';
import { ExternalLibraryName, ExternalLibraryNames } from './types/ExternalTypes';

import Material from '../../containers/material/MaterialContainer'; // TODO: Remove

export type ApplicationProps = DispatchProps & StateProps & RouteComponentProps<{}>;

export type DispatchProps = {
  handleClearContext: (
    chapter: number,
    variant: Variant,
    externalLibraryName: ExternalLibraryName
  ) => void;
  handleEditorValueChange: (val: string) => void;
  handleEditorUpdateBreakpoints: (breakpoints: string[]) => void;
  handleEnsureLibrariesLoaded: () => void;
  handleLogOut: () => void;
  handleExternalLibrarySelect: (external: ExternalLibraryName) => void;
  handleSetExecTime: (execTime: string) => void;
};

export type StateProps = {
  accessToken?: string;
  currentPlaygroundChapter: number;
  currentPlaygroundVariant: Variant;
  role?: Role;
  title: string;
  name?: string;
  currentExternalLibrary: ExternalLibraryName;
};

const assessmentRegExp = ':assessmentId(-?\\d+)?/:questionId(\\d+)?';

class Application extends React.Component<ApplicationProps, {}> {
  public componentDidMount() {
    parsePlayground(this.props);
  }

  public render() {
    return (
      <div className="Application">
        <NavigationBar
          handleLogOut={this.props.handleLogOut}
          role={this.props.role}
          name={this.props.name}
          title={this.props.title}
        />
        <div className="Application__main">
          <Switch>
            <Route path="/academy" component={toAcademy(this.props)} />
            <Route path={`/mission-control/${assessmentRegExp}`} render={toIncubator} />
            <Route path="/playground" component={Playground} />
            <Route path="/login" render={toLogin(this.props)} />
            <Route path="/contributors" component={Contributors} />
            <Route path="/material" component={Material} />
            <Route path="/sourcecast" component={SourcecastContainer} />
            <Route exact={true} path="/" render={this.redirectToPlayground} />
            <Route component={NotFound} />
          </Switch>
        </div>
      </div>
    );
  }

  private redirectToPlayground = () => <Redirect to="/playground" />;
}

/**
 * A user routes to /academy,
 *  1. If the user is logged in, render the Academy component
 *  2. If the user is not logged in, redirect to /login
 */
const toAcademy = (props: ApplicationProps) =>
  props.accessToken === undefined || props.role === undefined
    ? () => <Redirect to="/login" />
    : () => <Academy accessToken={props.accessToken} role={props.role!} />;

const toLogin = (props: ApplicationProps) => () => (
  <Login luminusCode={qs.parse(props.location.search).code} />
);

const parsePlayground = (props: ApplicationProps) => {
  const prgrm = parsePrgrm(props);
  const chapter = parseChapter(props) || props.currentPlaygroundChapter;
  const variant = parseVariant(props, chapter) || props.currentPlaygroundVariant;
  const externalLibraryName = parseExternalLibrary(props) || props.currentExternalLibrary;
  const execTime = parseExecTime(props);
  if (prgrm) {
    props.handleEditorValueChange(prgrm);
    props.handleEnsureLibrariesLoaded();
    props.handleClearContext(chapter, variant, externalLibraryName);
    props.handleExternalLibrarySelect(externalLibraryName);
    props.handleSetExecTime(execTime);
  }
};

const toIncubator = (routerProps: RouteComponentProps<any>) => <MissionControlContainer />;

const parsePrgrm = (props: RouteComponentProps<{}>) => {
  const qsParsed = qs.parse(props.location.hash);
  // legacy support
  const program = qsParsed.lz !== undefined ? qsParsed.lz : qsParsed.prgrm;
  return program !== undefined ? decompressFromEncodedURIComponent(program) : undefined;
};

const parseChapter = (props: RouteComponentProps<{}>) => {
  const chapQuery = qs.parse(props.location.hash).chap;
  // find a language with this chapter (if any)
  const langWithMatchingChapter = sourceLanguages.find(language => language.chapter === chapQuery);

  const chap: number = langWithMatchingChapter
    ? langWithMatchingChapter.chapter
    : chapQuery === undefined
    ? NaN
    : parseInt(chapQuery, 10);

  return chap ? chap : undefined;
};

const parseVariant = (props: RouteComponentProps<{}>, chap: number) => {
  const variantQuery = qs.parse(props.location.hash).variant;
  // find a language with this variant and chapter (if any)
  const matchingLang = sourceLanguages.find(
    language => language.chapter === chap && language.variant === variantQuery
  );

  const variant: Variant = matchingLang ? matchingLang.variant : 'default';

  return variant;
};

const parseExternalLibrary = (props: RouteComponentProps<{}>) => {
  const ext = qs.parse(props.location.hash).ext || '';
  return Object.values(ExternalLibraryNames).includes(ext) ? ext : ExternalLibraryNames.NONE;
};

const parseExecTime = (props: RouteComponentProps<{}>) => {
  const time = qs.parse(props.location.hash).exec || '1000';
  // Parse the time string to a number, defaulting execTime to 1000
  const execTime = stringParamToInt(time) || 1000;
  return `${execTime < 1000 ? 1000 : execTime}`;
};

export default Application;
