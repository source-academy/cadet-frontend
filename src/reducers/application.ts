import { Action, Reducer } from 'redux';
import { defaultApplication, IApplicationState } from './states';

export const reducer: Reducer<IApplicationState> = (state = defaultApplication, action: Action) => {
  return state;
};
