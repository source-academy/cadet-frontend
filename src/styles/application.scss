.Application {
    display: flex;
    flex-direction: column;
}

.Application__main {
    margin-top: 1rem;
    margin-left: 1rem;
}
