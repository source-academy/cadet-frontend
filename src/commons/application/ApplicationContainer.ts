import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
import { withRouter } from 'react-router';
import { bindActionCreators, Dispatch } from 'redux';

import { logOut } from './actions/CommonsActions';
import { loginGitHub, logoutGitHub } from './actions/SessionActions';
import Application, { DispatchProps, StateProps } from './Application';
import { OverallState } from './ApplicationTypes';

/**
 * Provides the title of the application for display.
 * An object with the relevant properties must be
 * returned instead of an object of type @type {IApplicationProps},
 * as the routing properties of @type {RouteComponentProps} are
 * provided using the withRouter() method below.
 */
const mapStateToProps: MapStateToProps<StateProps, {}, OverallState> = state => ({
  title: state.application.title,
  role: state.session.role,
  name: state.session.name
});

const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = (dispatch: Dispatch) =>
  bindActionCreators(
    { handleLogOut: logOut, handleGitHubLogIn: loginGitHub, handleGitHubLogOut: logoutGitHub },
    dispatch
  );

const ApplicationContainer = withRouter(connect(mapStateToProps, mapDispatchToProps)(Application));

export default ApplicationContainer;
