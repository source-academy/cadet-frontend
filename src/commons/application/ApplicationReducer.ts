import { Action, Reducer } from 'redux';
import { defaultApplication, IApplicationState } from 'src/commons/application/ApplicationTypes';

export const ApplicationReducer: Reducer<IApplicationState> = (
  state = defaultApplication,
  action: Action
) => {
  return state;
};
