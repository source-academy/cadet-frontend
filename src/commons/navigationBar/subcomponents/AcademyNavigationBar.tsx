import { Alignment, Classes, Icon, Navbar, NavbarGroup } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import classNames from 'classnames';
import * as React from 'react';
import { NavLink } from 'react-router-dom';

import { Role } from '../../application/ApplicationTypes';
import { AssessmentCategories } from '../../assessment/AssessmentTypes';
import NotificationBadgeContainer from '../../notificationBadge/NotificationBadgeContainer';
import { filterNotificationsByType } from '../../notificationBadge/NotificationBadgeHelper';
import { assessmentCategoryLink } from '../../utils/ParamParseHelper';

type OwnProps = {
  role: Role;
};

const AcademyNavigationBar: React.FunctionComponent<OwnProps> = props => (
  <Navbar className="NavigationBar secondary-navbar">
    <NavbarGroup align={Alignment.LEFT}>
      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Mission)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.FLAME} />
        <div className="navbar-button-text hidden-xs hidden-sm">Missions</div>
        <NotificationBadgeContainer
          notificationFilter={filterNotificationsByType(AssessmentCategories.Mission)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Sidequest)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.LIGHTBULB} />
        <div className="navbar-button-text hidden-xs hidden-sm">Quests</div>
        <NotificationBadgeContainer
          notificationFilter={filterNotificationsByType(AssessmentCategories.Sidequest)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Path)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.PREDICTIVE_ANALYSIS} />
        <div className="navbar-button-text hidden-xs hidden-sm">Paths</div>
        <NotificationBadgeContainer
          notificationFilter={filterNotificationsByType(AssessmentCategories.Path)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Contest)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.COMPARISON} />
        <div className="navbar-button-text hidden-xs hidden-sm">Contests</div>
        <NotificationBadgeContainer
          notificationFilter={filterNotificationsByType(AssessmentCategories.Contest)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Practical)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.MANUAL} />
        <div className="navbar-button-text hidden-xs hidden-sm">Others</div>
      </NavLink>
    </NavbarGroup>
    {props.role === Role.Admin || props.role === Role.Staff ? (
      <NavbarGroup align={Alignment.RIGHT}>
        <NavLink
          to={'/academy/groundcontrol'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon="satellite" />
          <div className="navbar-button-text hidden-xs hidden-sm">Ground Control</div>
        </NavLink>

        <NavLink
          to={'/academy/dashboard'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon="globe" />
          <div className="navbar-button-text hidden-xs hidden-sm">Dashboard</div>
        </NavLink>

        <NavLink
          to={'/academy/sourcereel'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.MOBILE_VIDEO} />
          <div className="navbar-button-text hidden-xs hidden-sm">Sourcereel</div>
        </NavLink>

        <NavLink
          to={'/academy/grading'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.ENDORSED} />
          <div className="navbar-button-text hidden-xs hidden-sm">Grading</div>
          <NotificationBadgeContainer
            notificationFilter={filterNotificationsByType('Grading')}
            disableHover={true}
          />
        </NavLink>

        <NavLink
          to={'/academy/storysimulator'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.CROWN} />
          <div className="navbar-button-text hidden-xs hidden-sm hidden-md">Story Simulator</div>
        </NavLink>
      </NavbarGroup>
    ) : null}
  </Navbar>
);

export default AcademyNavigationBar;
