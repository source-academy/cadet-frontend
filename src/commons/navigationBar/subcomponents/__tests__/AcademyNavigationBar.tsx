import { shallow } from 'enzyme';

import { Role } from '../../../application/ApplicationTypes';
import AcademyNavigationBar from '../AcademyNavigationBar';

test('MissionControl, GroundControl, Sourcereel, StorySimulator, Dashboard, Grading and AdminPanel NavLinks do NOT render for Role.Student', () => {
  const props = {
    role: Role.Student,
    notifications: [],
    assessmentTypes: ['Missions', 'Quests', 'Paths', 'Contests', 'Others']
  };
  const tree = shallow(<AcademyNavigationBar {...props} />);
  expect(tree.debug()).toMatchSnapshot();
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/grading' }).exists())
  ).toHaveLength(0);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/groundcontrol' }).exists())
  ).toHaveLength(0);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/sourcereel' }).exists())
  ).toHaveLength(0);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/storysimulator' }).exists())
  ).toHaveLength(0);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/dashboard' }).exists())
  ).toHaveLength(0);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/adminpanel' }).exists())
  ).toHaveLength(0);
});

test('MissionControl, GroundControl, Sourcereel, StorySimulator, Dashboard and Grading NavLinks render for Role.Staff', () => {
  const props = {
    role: Role.Staff,
    notifications: [],
    assessmentTypes: ['Missions', 'Quests', 'Paths', 'Contests', 'Others']
  };
  const tree = shallow(<AcademyNavigationBar {...props} />);
  expect(tree.debug()).toMatchSnapshot();
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/grading' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/groundcontrol' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/sourcereel' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/storysimulator' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/dashboard' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/adminpanel' }).exists())
  ).toHaveLength(0);
});

test('MissionControl, GroundControl, Sourcereel, StorySimulator, Dashboard, Grading and AdminPanel NavLinks render for Role.Admin', () => {
  const props = {
    role: Role.Admin,
    notifications: [],
    assessmentTypes: ['Missions', 'Quests', 'Paths', 'Contests', 'Others']
  };
  const tree = shallow(<AcademyNavigationBar {...props} />);
  expect(tree.debug()).toMatchSnapshot();
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/grading' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/groundcontrol' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/sourcereel' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/storysimulator' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/dashboard' }).exists())
  ).toHaveLength(1);
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/adminpanel' }).exists())
  ).toHaveLength(1);
});
