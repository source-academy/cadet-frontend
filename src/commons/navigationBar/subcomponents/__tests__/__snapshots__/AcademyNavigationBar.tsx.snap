// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionControl, GroundControl, Sourcereel, StorySimulator, Dashboard and Grading NavLinks render for Role.Staff 1`] = `
"<Blueprint3.Navbar className=\\"NavigationBar secondary-navbar\\">
  <Blueprint3.NavbarGroup align=\\"left\\">
    <NavLink to=\\"/academy/missions\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Missions
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/quests\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"lightbulb\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Quests
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/paths\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"predictive-analysis\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Paths
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/contests\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"comparison\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Contests
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/others\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"manual\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Others
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
  </Blueprint3.NavbarGroup>
  <Blueprint3.NavbarGroup align=\\"right\\">
    <NavLink to=\\"/academy/groundcontrol\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"satellite\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Ground Control
      </div>
    </NavLink>
    <NavLink to=\\"/academy/dashboard\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"globe\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Dashboard
      </div>
    </NavLink>
    <NavLink to=\\"/academy/sourcereel\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"mobile-video\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm hidden-md\\">
        Sourcereel
      </div>
    </NavLink>
    <NavLink to=\\"/academy/grading\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"endorsed\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm hidden-md\\">
        Grading
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/storysimulator\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"crown\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm hidden-md\\">
        Story Simulator
      </div>
    </NavLink>
  </Blueprint3.NavbarGroup>
</Blueprint3.Navbar>"
`;

exports[`MissionControl, GroundControl, Sourcereel, StorySimulator, Dashboard, Grading and AdminPanel NavLinks do NOT render for Role.Student 1`] = `
"<Blueprint3.Navbar className=\\"NavigationBar secondary-navbar\\">
  <Blueprint3.NavbarGroup align=\\"left\\">
    <NavLink to=\\"/academy/missions\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Missions
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/quests\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"lightbulb\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Quests
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/paths\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"predictive-analysis\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Paths
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/contests\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"comparison\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Contests
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/others\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"manual\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Others
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
  </Blueprint3.NavbarGroup>
</Blueprint3.Navbar>"
`;

exports[`MissionControl, GroundControl, Sourcereel, StorySimulator, Dashboard, Grading and AdminPanel NavLinks render for Role.Admin 1`] = `
"<Blueprint3.Navbar className=\\"NavigationBar secondary-navbar\\">
  <Blueprint3.NavbarGroup align=\\"left\\">
    <NavLink to=\\"/academy/missions\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Missions
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/quests\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"lightbulb\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Quests
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/paths\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"predictive-analysis\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Paths
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/contests\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"comparison\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Contests
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/others\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"manual\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Others
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
  </Blueprint3.NavbarGroup>
  <Blueprint3.NavbarGroup align=\\"right\\">
    <NavLink to=\\"/academy/groundcontrol\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"satellite\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Ground Control
      </div>
    </NavLink>
    <NavLink to=\\"/academy/dashboard\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"globe\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Dashboard
      </div>
    </NavLink>
    <NavLink to=\\"/academy/sourcereel\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"mobile-video\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm hidden-md\\">
        Sourcereel
      </div>
    </NavLink>
    <NavLink to=\\"/academy/grading\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"endorsed\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm hidden-md\\">
        Grading
      </div>
      <Connect(NotificationBadge) notificationFilter={[Function (anonymous)]} disableHover={true} />
    </NavLink>
    <NavLink to=\\"/academy/storysimulator\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"crown\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm hidden-md\\">
        Story Simulator
      </div>
    </NavLink>
    <NavLink to=\\"/academy/adminpanel\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"settings\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm hidden-md hidden-lg\\">
        Admin Panel
      </div>
    </NavLink>
  </Blueprint3.NavbarGroup>
</Blueprint3.Navbar>"
`;
